superagi/controllers/agent.py

```diff
@@ -136,15 +141,15 @@ def update_agent(agent_id: int, agent: AgentIn,
             HTTPException (Status Code=404): If the Agent or associated Project is not found.
     """
 
-    db_agent = db.session.query(Agent).filter(Agent.id == agent_id).first()
+    db_agent = db.session.query(Agent).filter(Agent.id == agent_id, Agent.is_deleted == False).first()
     if not db_agent:
         raise HTTPException(status_code=404, detail="agent not found")
 
     if agent.project_id:
-        project = db.session.query(Project).get(agent.project_id)
-        if not project:
+        if project := db.session.query(Project).get(agent.project_id):
+            db_agent.project_id = project.id
+        else:
```

Comment:
why is this changed, any specific reason?

Reply:
just refactored it, reduces lines of code. thought will make it more pythonic